vic/drivers/classic/src/read_initial_model_state.c

```diff
@@ -688,4 +692,90 @@ read_initial_model_state(FILE            *init_state,
             }
         }
     }
+
+    // Check that soil moisture does not exceed maximum allowed
+    for (veg = 0; veg <= Nveg; veg++) {
+        for (band = 0; band < Nbands; band++) {
+            for (lidx = 0; lidx < options.Nlayer; lidx++) {
+                if (cell[veg][band].layer[lidx].moist >
+                    soil_con->max_moist[lidx]) {
+                    log_warn("Initial soil moisture (%f mm) exceeds "
+                             "maximum (%f mm) in layer %zu for veg tile "
+                             "%d and snow band %d.  Resetting to maximum.",
+                             cell[veg][band].layer[lidx].moist,
+                             soil_con->max_moist[lidx], lidx, veg, band);
+                    for (frost_area = 0;
+                         frost_area < options.Nfrost;
+                         frost_area++) {
+                        cell[veg][band].layer[lidx].ice[frost_area] *=
+                            soil_con->max_moist[lidx] /
+                            cell[veg][band].layer[lidx].moist;
+                    }
+                    cell[veg][band].layer[lidx].moist =
+                        soil_con->max_moist[lidx];
+                }
+
+                for (frost_area = 0;
+                     frost_area < options.Nfrost;
+                     frost_area++) {
+                    if (cell[veg][band].layer[lidx].ice[frost_area] >
+                        cell[veg][band].layer[lidx].moist) {
+                        cell[veg][band].layer[lidx].ice[frost_area] =
+                            cell[veg][band].layer[lidx].moist;
+                    }
+                }
+            }
+        }
+
+        // Override possible bad values of soil moisture under lake coming from state file
+        // (ideally we wouldn't store these in the state file in the first place)
+        if (options.LAKES && (int) veg == lake_con.lake_idx) {
+            for (lidx = 0; lidx < options.Nlayer; lidx++) {
+                lake_var->soil.layer[lidx].moist =
+                    soil_con->max_moist[lidx];
+                for (frost_area = 0;
+                     frost_area < options.Nfrost;
+                     frost_area++) {
+                    if (lake_var->soil.layer[lidx].ice[frost_area] >
+                        lake_var->soil.layer[lidx].moist) {
+                        lake_var->soil.layer[lidx].ice[frost_area] =
+                            lake_var->soil.layer[lidx].moist;
+                    }
+                }
+            }
+        }
+    }
+
+    // Check that snow pack terms are self-consistent
```

Comment:
I'm inclined to remove this section of the code.  The cap on soil moisture storage seems more valuable to me, since during calibration the soil moisture storage capacity is commonly modified, which could lead to errors if we didn't do that cap.  But to cap snowpack liquid moisture storage seems to be too much - snowpack liquid water storage capacity is not a commonly-calibrated parameter - if we accept this adjustment, then why not cap all other state variables too?  So, I'll remove it.